### PR TITLE
PYTHON 3920 - Migrate AWS Auth Tests to use AWS Secrets

### PR DESCRIPTION
**Pull request by NoahStapp**

### CODE REVIEW

File: .evergreen/auth_aws/aws_tester.py [hunk 1]
```diff
@@ -23,12 +22,25 @@
...
 AWS_ACCOUNT_ARN = "arn:aws:sts::557821124784:assumed-role/evergreen_task_hosts_instance_role_production/*"
-
-
-with open(os.path.join(HERE, 'aws_e2e_setup.json')) as fid:
-   CONFIG = json.load(fid)
-
-
+_USE_AWS_SECRETS = False
+
+if "USE_AWS_SECRETS" not in os.environ:
```
Comment (blink1073) [inline .evergreen/auth_aws/aws_tester.py:27]: Maybe instead we try to load from env if there is no `aws_e2e_setup.json` file?

Comment (NoahStapp) [reply .evergreen/auth_aws/aws_tester.py:27]: I like that idea, but what happens in the edge case where there is an `aws_e2e_setup.json` file but we want to load from env anyway?

File: .evergreen/auth_aws/util.py [hunk 2]
```diff
@@ -0,0 +1,5 @@
+def get_key(key: str, uppercase: bool) -> str:
```
Comment (blink1073) [inline .evergreen/auth_aws/util.py:1]: I think this file should be in `lib`

**Review by blink1073 — CHANGES_REQUESTED** [review]

File: .evergreen/auth_aws/aws_tester.py [hunk 3]
```diff
@@ -7,26 +7,30 @@
...
 from aws_assume_web_role import _assume_role_with_web_identity
 from aws_assign_instance_profile import _assign_instance_policy
 
-ASSUMED_ROLE = "arn:aws:sts::557821124784:assumed-role/authtest_user_assume_role/*";
+ASSUMED_ROLE = "arn:aws:sts::557821124784:assumed-role/authtest_user_assume_role/*"
 ASSUMED_WEB_ROLE = "arn:aws:sts::857654397073:assumed-role/webIdentityTestRole/*"
 
 # This varies based on hosting EC2 as the account id and role name can vary
 AWS_ACCOUNT_ARN = "arn:aws:sts::557821124784:assumed-role/evergreen_task_hosts_instance_role_production/*"
+_USE_AWS_SECRETS = False
```
[the remark this answers is not in the document]
Comment (NoahStapp) [reply .evergreen/auth_aws/aws_tester.py:26]: I like it, good idea.

**Comment by blink1073** [comment]
Before merging, can you please run a patch build that uses the prior AWS secrets handling in pymongo and this branch?

**Comment by NoahStapp** [comment]
> Before merging, can you please run a patch build that uses the prior AWS secrets handling in pymongo and this branch?

You mean without using the AWS secret vaults in pymongo but still using this branch?

**Comment by blink1073** [comment]
> You mean without using the AWS secret vaults in pymongo but still using this branch?

I mean pymongo master, but targeting your Drivers Evergreen Tools branch to make sure it still works with the old methods.

**Comment by NoahStapp** [comment]
> > You mean without using the AWS secret vaults in pymongo but still using this branch?
> 
> I mean pymongo master, but targeting your Drivers Evergreen Tools branch to make sure it still works with the old methods.

https://spruce.mongodb.com/version/64f75ec0850e61b528e01d27/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC successfully passed!

**Review by blink1073 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**